### PR TITLE
add route poking for solaris and bsd

### PR DESCRIPTION
**Pull request by despair86**

### CODE REVIEW

**Comment by despair86** [comment]
performance is disappointing cause i literally cannot run the optimiser on libzmq or else lokinet blows up in my face (waiting on clang-11 and g++ 11 lul)

File: llarp/net/route.cpp [hunk 1]
```diff
@@ -449,14 +461,18 @@ namespace llarp::net
...
     LogDebug("get gateways not on ", ifname);
     const auto maybe = GetIFAddr(ifname);
     if (not maybe.has_value())
       return gateways;
     const auto interface = maybe->toString();
     // mac os is so godawful man
+#if __APPLE__
     FILE* p = popen("/usr/sbin/netstat -rn -f inet", "r");
+#else
+    FILE* p = popen("/usr/bin/netstat -rn -f inet", "r");
```
Comment (despair86) [inline llarp/net/route.cpp:491]: being a mashup of BSD and SysV works well here: macOS and Solaris have the same flavour of `netstat(1M)`

File: llarp/net/route.cpp [hunk 2]
```diff
@@ -408,6 +416,10 @@ namespace llarp::net
     Execute("/sbin/route -n delete -cloning -net 0.0.0.0 -netmask 128.0.0.0 -interface " + ifname);
     Execute(
         "/sbin/route -n delete -cloning -net 128.0.0.0 -netmask 128.0.0.0 -interface " + ifname);
+#elif __sun
+    const auto iface = GetIFAddr(ifname);
+    Execute("/usr/sbin/route delete 0.0.0.0/1 -iface " + iface->toHost());
```
Comment (despair86) [inline llarp/net/route.cpp:436]: i could have also recycled the route commands but `-cloning` does nothing on SysV

Comment (majestrate) [reply llarp/net/route.cpp:436]: this returns a std::optional and de-referencing a std::optional is undefined behavior

Comment (despair86) [reply llarp/net/route.cpp:436]: then what the hell is line 378
```
 read_addr(maybe->toHost().c_str(), &gw_addr);
```

**Comment by despair86** [comment]
this is ready now since the SIIT fixes were stuffed into #1421

**Comment by despair86** [comment]
route poker runs twice when exit is enabled: lokimq via lokinet-vpn triggers the original route poker which works correctly
then the new route poker is triggered and fails with error `[route|file] exists` which is harmless

new route poker leaks routes on BSD and possibly Macintosh (they use the same command to add and delet individual routes). removing the /1 routes works as expected however.

output of old route poker at exit:
```
delete route 223.223.223.223 via 223.223.223.223
```
behaviour of new route poker is inconsistent on BSD, and mostly steps out of the way on Solaris oddly enough (routes are not leaked on exit, at least)

last but not least I still haven't figured out how to get udp sockets from timing out so quickly and causing tunnel layer tcp timeouts (bsd only)

**Comment by majestrate** [comment]
i would really rather use a routing socket at this point.

**Comment by despair86** [comment]
i dropped clang-9 a long time ago so all lints fail for now

**Comment by despair86** [comment]
finally got rid of windows hot loop allocation :DDD

File: llarp/ev/ev_win32.cpp [hunk 3]
```diff
@@ -12,6 +12,12 @@ static HANDLE tun_event_queue = INVALID_HANDLE_VALUE;
 static HANDLE* kThreadPool;
 static int poolSize;
 
+// ring buffer to replace dynamic loop allocation
+// we only care about the first dimension of the array
+static byte_t pkt_ring[1000][1500];
```
Comment (majestrate) [inline llarp/ev/ev_win32.cpp]: can we put this into another PR?

**Comment by majestrate** [comment]
we are gutting this code soon so i think i'll be closing this.
